### DISCUSSION TITLE
Use occluders for performance improvement

### DISCUSSION THREAD

**Discussion by elvisish** [codecat/godot-tbloader#106] [Ideas]
Here's the issue/proposal: https://github.com/codecat/godot-tbloader/issues/101
I haven't the knowledge or experience, but it would be great for performance if TBLoader could automagically build occlusion meshes for the world geometry.

**Comment by solitaryurt** [comment]
You'd need to split the map into multiple `func_group`'s. If you're using just one large `world_spawn` you'll have one `MeshInstance3D` for your map, and as a result it will never cull when occluded (it will never occlude).

Splitting into multiple `func_group`'s will allow for portions of the map to occlude and therefore cull.

**Reply by elvisish** [reply]
> You'd need to split the map into multiple `func_group`'s. If you're using just one large `world_spawn` you'll have one `MeshInstance3D` for your map, and as a result it will never cull when occluded (it will never occlude).
> 
> Splitting into multiple `func_group`'s will allow for portions of the map to occlude and therefore cull.

Is that how qodot/func_godot handles it? I thought it just added many occluders based on the mesh brushes (although I could be wrong, I haven't used it much),

**Reply by solitaryurt** [reply]
Last I tried func_godot, it was splitting each brush into a `MeshInstance3D` node.
That would correctly cull each brush that was occluded, but there is a heavy performance cost depending on the size of your map.
